src/parquet/arrow/writer.cc

```diff
@@ -106,21 +106,19 @@ class LevelBuilder {
   NOT_IMPLEMENTED_VISIT(Struct)
   NOT_IMPLEMENTED_VISIT(Union)
   NOT_IMPLEMENTED_VISIT(Dictionary)
-  NOT_IMPLEMENTED_VISIT(Interval)
 
   Status GenerateLevels(const Array& array, const std::shared_ptr<Field>& field,
-                        int64_t* values_offset, ::arrow::Type::type* values_type,
-                        int64_t* num_values, int64_t* num_levels,
-                        std::shared_ptr<Buffer>* def_levels,
-                        std::shared_ptr<Buffer>* rep_levels,
+                        int64_t* values_offset, int64_t* num_values, int64_t* num_levels,
+                        const std::shared_ptr<PoolBuffer>& def_levels_scratch,
```

Comment:
I don't really understand the need for `def_levels_scratch` here. Can you give me some context why this is needed?

Reply:
In the nested case, the definition levels are generated using a builder. Otherwise we can reuse the same pre-allocated memory for the levels. This could all be refactored potentially to reuse a common shared buffer, maybe we can do that later